test/controllers/orders_controller_test.rb

```diff
@@ -0,0 +1,57 @@
+require 'test_helper'
+
+class OrdersControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    @order = orders(:one)
+  end
+
+  test "should get index" do
+    get orders_url :format => :json
+    assert_response :success
+    response = JSON.parse(@response.body)
```

Comment:
kind of a naming nitpick here, but your local variable `response` isn't exactly accurate for what it is. A better name here might be `response_body`, or `body`.

I'm also pretty sure that in this scope you'd be able to access `response` instead of `@response`. When I see an instance variable I try to go looking for where it's initialized, but in this case it's available because of the `get` request. Using `response` as a local variable helps make it clear that this is something scoped to the specific test, imo. If it's not accessible using `response` then disregard :)

Reply:
Coolio, will check this out - got it from an example online...